### PR TITLE
chore(docs): reformat and update technology stack documentation

### PR DESCRIPTION
**Pull request by debloper**
A rendered version of this edit can be found on: https://github.com/debloper/fabric8-planner/blob/master/docs/technology_stack.adoc

### CODE REVIEW

**Comment by fabric8cd** [comment]
@debloper snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-3 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

File: docs/technology_stack.adoc [hunk 1]
```diff
@@ -7,35 +7,42 @@
...
+The Planner is an Angular app built on top of Node.js. It can run as a standalone app, or be used as a library to be integrated into the link:./fabric8-platform[Fabric8 Platform].
 
-== Angular 4
-<TBD>
+Development or deployment of Planner is fairly platform agnostic, so we can skip the operating system requirements and discuss from the runtime environment to way-up.
 
-== CSS and SASS
-<WIP, to be changed to PF CSS, Less>
+== Development Runtime
+Planner uses *`Node.js`* as the platform-runtime, on top of which it builds; on that line, we use *`npm`* for resolving and distributing packages for the project.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: Imho, 'on top of which it builds;' 'on that line' these parts are not very clear.
Can we reword it to make it simpler? How about, 'Planner uses *`Node.js`* for building the platform-runtime and uses *`npm`* for resolving and distributing (can we mention here which packages?)packages for the project.' wdyt?
As per convention use bold for software packages *npm*

File: docs/technology_stack.adoc [hunk 2]
```diff
@@ -7,35 +7,42 @@
...
 
-Fabric8-planner uses SASS for it's stylesheets. It also uses the Angular emulation
-of the shadow dom, hence it is recommended to place your styles in the
-*_.component.scss_* file next to the HTML and the TypeScript.
+We keep ourselves compatible with the latest stable releases of *`Node.js`* & *`npm`* - however, *`Node.js >= 6.xx`* & *`npm >= 3.xx`* releases should work and are recommended.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: The words 'should work and' can be dropped.

File: docs/technology_stack.adoc [hunk 3]
```diff
@@ -7,35 +7,42 @@
...
 
-If you want to create a shared style that could be used by multiple components, then we recommend adding it as a mixin to
-*_src/assets/stylesheets/_planner-mixins.scss_*. The mixins are imported into every
-*_.component.scss_* file.
+== Development Pipeline
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: In the introduction of this section, I suggest we have a one liner explaining when this section comes into the development picture and a line about the subsections in this sections such as task runner, build system etc.

File: docs/technology_stack.adoc [hunk 4]
```diff
@@ -7,35 +7,42 @@
...
+Jenkins
+
+Release Framework::
+semantic-release
+
+Deployment::
+Docker, nginx
+
+== Application Framework
+We use *`Angular 4`* to develop Planner. Main application code resides in `./src`, in which the `./src/app` contains all the application logics, subdivided into specific functional groups.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: As per convention, names of directory, file names etc should be in Italics and bold, `./src` > *_./src_*
logics>logic

File: docs/technology_stack.adoc [hunk 5]
```diff
@@ -7,35 +7,42 @@
...
+semantic-release
+
+Deployment::
+Docker, nginx
+
+== Application Framework
+We use *`Angular 4`* to develop Planner. Main application code resides in `./src`, in which the `./src/app` contains all the application logics, subdivided into specific functional groups. 
+
+== Programming Languages
+For development ease, we dont use HTML, CSS, JS as is, but Angular Templates, Less and TypeScript instead.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: don't> do not

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-5 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-6 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-7 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

File: docs/technology_stack.adoc (hunk 1 above)
Comment (debloper) [inline docs/technology_stack.adoc]: Actually here the platform-runtime is completely different from what we generally term the platform (Fabric8-UI) or runtime (the minimal frontend for standalone Planner). In this case platoform runtime is a piece of software that enables a language to interface with the OS (in this case the language is JS and the platform-runtime is Node.js). Sorry for the confusion, but there's barely any way around as those are the apt terms here.

`*uses Node.js* as the platform-runtime` and `uses Node.js for building the platform-runtime` are completely different things, and the later isn't intended here (we're not building Node.js, we're using it for making JS interact with the platform, which can be Linus, Windows, OSX etc.)

Also, the packages resolved are a long list and is already available inside `package.json` (standard process for Node,js apps), don't think we need to list hundreds of package names here that's prone to changes often.

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-8 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-10 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-12 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

File: docs/technology_stack.adoc [hunk 6]
```diff
@@ -7,35 +7,43 @@
...
+The Planner is an Angular app built on top of Node.js. It can run as a standalone app, or be used as a library to be integrated into the link:./fabric8-ui[Fabric8 UI].
 
-== Angular 4
-<TBD>
+Development or deployment of Planner is fairly platform agnostic, so we can skip the operating system requirements and discuss from the runtime environment onward.
 
-== CSS and SASS
-<WIP, to be changed to PF CSS, Less>
+== Development Runtime
+Planner uses *`Node.js`* as the platform-runtime, on top of which it builds; and use *`npm`* for resolving and distributing packages for the project.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: use *`npm`*> uses *`npm`* 
We could put it this way too, 
Planner uses:
- `Node.js`* as the platform-runtime, on top of which it builds (the runtime component?)
- *`npm`* for resolving and distributing packages for the project

File: docs/technology_stack.adoc [hunk 7]
```diff
@@ -7,35 +7,43 @@
...
+== Development Pipeline
 
-You can then create a real class by doing something like:
+Following are the components or parts of the process followed in order to develop, build, test and deploy Planner. Most of these components are pretty standard, with limited project specific customizations/configurations applied.
 
-    .my-class {
-      @include my-class;
-    }
+Task Runner::
+We use Gulp for running predefined tasks. These tasks include, compiling/preprocessing the LESS/TypeScript sources into CSS/JS, building the distributable artifacts, running dev-server in watch mode and general house-keeping (install, reinstall, cleanup etc.).
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: Names of all Software packages, like Gulp, should be in bold and use the same case(as per their standard), every time it is referenced. As in OpenShift has to be written this way.

File: docs/technology_stack.adoc [hunk 8]
```diff
@@ -7,35 +7,43 @@
 toc::[]
 
 == Overview
-The following sections describe the technology stack used by Planner.
+The Planner is an Angular app built on top of Node.js. It can run as a standalone app, or be used as a library to be integrated into the link:./fabric8-ui[Fabric8 UI].
```
Comment (nimishamukherjee) [inline docs/technology_stack.adoc]: Planner can be used as a library by any app. Platform UI is one of them.

File: docs/technology_stack.adoc [hunk 9]
```diff
@@ -7,35 +7,43 @@
...
-*_.component.scss_* file next to the HTML and the TypeScript.
+We keep ourselves compatible with the latest stable releases of *`Node.js`* & *`npm`* - however, *`Node.js >= 6.xx`* & *`npm >= 3.xx`* releases are recommended.
 
-If you want to create a shared style that could be used by multiple components, then we recommend adding it as a mixin to
-*_src/assets/stylesheets/_planner-mixins.scss_*. The mixins are imported into every
-*_.component.scss_* file.
+== Development Pipeline
 
-You can then create a real class by doing something like:
+Following are the components or parts of the process followed in order to develop, build, test and deploy Planner. Most of these components are pretty standard, with limited project specific customizations/configurations applied.
```
Comment (nimishamukherjee) [inline docs/technology_stack.adoc]: Remove pretty

File: docs/technology_stack.adoc [hunk 10]
```diff
@@ -7,35 +7,43 @@
...
-    .my-class {
-      @include my-class;
-    }
+Task Runner::
+We use Gulp for running predefined tasks. These tasks include, compiling/preprocessing the LESS/TypeScript sources into CSS/JS, building the distributable artifacts, running dev-server in watch mode and general house-keeping (install, reinstall, cleanup etc.).
 
-We use mixins to avoid polluting components with unnecessary style classes, and to avoid
-an explosion of shared files.
+Build System::
+We use both Gulp and webpack for building the source. While gulp takes care of the library builds, webpack is used for building the (minimal) `*_./runtime_*` to use the library, which makes Planner work as a standalone app. The build system works as intended for now, but can be expanded to use more sophisticated switches to build according to different needs (dev, tests, prod etc.).
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: Remove period within parenthesis.

File: docs/technology_stack.adoc [hunk 11]
```diff
@@ -7,35 +7,43 @@
...
 
-The *_src/assets/stylesheets/_* directory includes a *_shared_* directory. These are
-shared global styles which will be refactored into a shared library at some point.
-You can update these styles only if you are making a truly global style, and are going to
-synchronize your changes across all the various UI projects.
+Test Framework::
+Our test stack is built with Karma, Protractor and PhantomJS. PhantomJS gives the headless browser engine to run the app, which Protractor interacts with as a WebDriverJS wrapper and runs the tests according to Karma's configurations.
```
Comment (nimishamukherjee) [inline docs/technology_stack.adoc]: Can you check with Len or Ruchir. I don't think we are using PhantomJS.

File: docs/technology_stack.adoc [hunk 12]
```diff
@@ -7,35 +7,43 @@
...
 
-== Gulp
-<TBD>
+Each `*_filename.ts_*` file is supposed to have unit tests along with them as a `*_filename.spec.ts_*` in the same parent directory. The functional tests reside inside `*_./runtime/tests/_*`. For more details about running the tests, please look into link:./testing.adoc[Testing Documentation].
```
Comment (nimishamukherjee) [inline docs/technology_stack.adoc]: Functional tests reside under runtime/src/tests

File: docs/technology_stack.adoc (hunk 11 above)
Comment (Preeticp) [inline docs/technology_stack.adoc]: Names of Softwares-bold.
Split into shorter sentences. Suggest:
PhantomJS provides the headless browser engine to run the app. Protractor interacts with it as a WebDriverJS wrapper and runs the tests according to Karma's configurations.

File: docs/technology_stack.adoc [hunk 13]
```diff
@@ -7,35 +7,43 @@
...
+
+CI/CD::
+The CI/CD pipeline is made with Jenkins, as defined in the `*_./Jenkinsfile_*` at the root of the repo. It uses the content of `*_./deploy_*` (more specifically, `*_./deploy/release.groovy_*`) to orcastrate the CI/CD.
+
+Release Framework::
+We use semantic-release for versioning and publishing our Planner library to npm registry. It's fairly configuration less CLI task (`$ npm run semantic-release`) which takes the latest PR/merges and commit messages to account. It generates the new version number, publishes to npm registry, and then generates the release tag on GitHub in one simple process.
+
+Deployment::
+At the end of the day Planner is a static library, and can be used/served with minimal static frontend runtime as a standalone app or with a full fledged Fabric8 UI. So whatever can make those setups run, can essentially be the deployment platform for Planner. We use Docker Compose to generate the image, Docker for running the built app container, and nginx as the web server to serve the app contents.
```
Comment (nimishamukherjee) [inline docs/technology_stack.adoc]: Do we need all this  - "At the end of the day Planner is a static library, and can be used/served with minimal static frontend runtime as a standalone app or with a full fledged Fabric8 UI. So whatever can make those setups run, can essentially be the deployment platform for Planner."

File: docs/technology_stack.adoc [hunk 14]
```diff
@@ -7,35 +7,43 @@
...
-synchronize your changes across all the various UI projects.
+Test Framework::
+Our test stack is built with Karma, Protractor and PhantomJS. PhantomJS gives the headless browser engine to run the app, which Protractor interacts with as a WebDriverJS wrapper and runs the tests according to Karma's configurations.
 
-== Gulp
-<TBD>
+Each `*_filename.ts_*` file is supposed to have unit tests along with them as a `*_filename.spec.ts_*` in the same parent directory. The functional tests reside inside `*_./runtime/tests/_*`. For more details about running the tests, please look into link:./testing.adoc[Testing Documentation].
+
+CI/CD::
+The CI/CD pipeline is made with Jenkins, as defined in the `*_./Jenkinsfile_*` at the root of the repo. It uses the content of `*_./deploy_*` (more specifically, `*_./deploy/release.groovy_*`) to orcastrate the CI/CD.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: orcastrate>orchestrate

File: docs/technology_stack.adoc [hunk 15]
```diff
@@ -7,35 +7,43 @@
...
+
+Deployment::
+At the end of the day Planner is a static library, and can be used/served with minimal static frontend runtime as a standalone app or with a full fledged Fabric8 UI. So whatever can make those setups run, can essentially be the deployment platform for Planner. We use Docker Compose to generate the image, Docker for running the built app container, and nginx as the web server to serve the app contents.
+
+== Application Framework
+We use *`Angular 4`* to develop Planner. Main application code resides in `*_./src_*`, in which the `*_./src/app_*` contains all the application logic, subdivided into specific functional groups. For the minimalistic runtime added to make Planner work standalone, the source lies inside `*_./runtime/src/app_*` which is also an *`Angular 4`* app in itself.
+
+== Programming Languages
+For development ease, we do not use HTML, CSS, JS as is, but Angular Templates, Less and TypeScript instead.
+
```
Comment (nimishamukherjee) [inline docs/technology_stack.adoc]: Mention PatternFly

File: docs/technology_stack.adoc [hunk 16]
```diff
@@ -7,35 +7,43 @@
...
 
-== Gulp
-<TBD>
+Each `*_filename.ts_*` file is supposed to have unit tests along with them as a `*_filename.spec.ts_*` in the same parent directory. The functional tests reside inside `*_./runtime/tests/_*`. For more details about running the tests, please look into link:./testing.adoc[Testing Documentation].
+
+CI/CD::
+The CI/CD pipeline is made with Jenkins, as defined in the `*_./Jenkinsfile_*` at the root of the repo. It uses the content of `*_./deploy_*` (more specifically, `*_./deploy/release.groovy_*`) to orcastrate the CI/CD.
+
+Release Framework::
+We use semantic-release for versioning and publishing our Planner library to npm registry. It's fairly configuration less CLI task (`$ npm run semantic-release`) which takes the latest PR/merges and commit messages to account. It generates the new version number, publishes to npm registry, and then generates the release tag on GitHub in one simple process.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: It's> It is a, to account>into account
PR/merges>PR, merges
Global: substitute / with 'or'

File: docs/technology_stack.adoc (hunk 13 above)
Comment (Preeticp) [inline docs/technology_stack.adoc]: with>as
So whatever can make those setups run, suggest this to be replaced with exact terms. 'Whatever' is very casual, maybe something like - So any backend application/software which can support it, ...
We use Docker... can start on a new para.
Replace app with application

**Comment by Preeticp** [comment]
@debloper I will be doing a few more rounds of Doc work. A lot of my comments are to do with formatting and consistency and other doc standards. So you could take care of the technical parts, I can take up the doc consistency part in my work.

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-14 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-15 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-17 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-18 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

File: docs/technology_stack.adoc (hunk 13 above)
Comment (debloper) [inline docs/technology_stack.adoc]: Included this part to clarify that planner doesn't have any added burden over the runtime's deployment setup that it's integrated into. Can be removed, but I think that's an important context to set for someone going through the doc, otherwise it seems like the mentioned requirements are extra parts and pieces (and not something that's already needed for the runtime/base-app).

Comment (debloper) [inline docs/technology_stack.adoc]: Not sure which `with > as` change is needed here that'll continue to have the same meaning.
Changed the other suggestions (in upcoming patch).

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-20 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-21 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

File: docs/technology_stack.adoc [hunk 17]
```diff
@@ -7,35 +7,45 @@
...
 
-== CSS and SASS
-<WIP, to be changed to PF CSS, Less>
+== Development Runtime
+Planner uses *`Node.js`* as the platform-runtime, on top of which it builds; and *`npm`* for resolving and distributing packages for the project.
 
-Fabric8-planner uses SASS for it's stylesheets. It also uses the Angular emulation
-of the shadow dom, hence it is recommended to place your styles in the
-*_.component.scss_* file next to the HTML and the TypeScript.
+We keep ourselves compatible with the latest stable releases of *`Node.js`* & *`npm`* - however, *`Node.js >= 6.xx`* & *`npm >= 3.xx`* releases are recommended.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: Do not use &>and

File: docs/technology_stack.adoc [hunk 18]
```diff
@@ -7,35 +7,45 @@
...
 
-If you want to create a shared style that could be used by multiple components, then we recommend adding it as a mixin to
-*_src/assets/stylesheets/_planner-mixins.scss_*. The mixins are imported into every
-*_.component.scss_* file.
+== Development Pipeline
 
-You can then create a real class by doing something like:
+Following are the components or parts of the process followed in order to develop, build, test and deploy *Planner*. Most of these components are standard, with limited project specific customizations and configurations applied.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: Suggestion: Replace Following with, This section lists/discusses

File: docs/technology_stack.adoc [hunk 19]
```diff
@@ -7,35 +7,45 @@
...
 
-    .my-class {
-      @include my-class;
-    }
+Task Runner::
+We use *Gulp* for running predefined tasks. These tasks include, compiling or preprocessing the LESS/TypeScript sources into CSS/JS, building the distributable artifacts, running dev-server in watch mode and general house-keeping (install, reinstall, cleanup etc).
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: It would be nice to use a bulleted list here.
These tasks include:
- compiling or preprocessing the LESS/TypeScript sources into CSS/JS 
- building the distributable artifacts
- running dev-server in watch mode 
- general house-keeping (install, reinstall, cleanup etc)

File: docs/technology_stack.adoc [hunk 20]
```diff
@@ -7,35 +7,45 @@
...
 
-We use mixins to avoid polluting components with unnecessary style classes, and to avoid
-an explosion of shared files.
+Build System::
+We use both *Gulp* and webpack for building the source. While *Gulp* takes care of the library builds, webpack is used for building the (minimal) `*_./runtime_*` to use the library, which makes *Planner* work as a standalone app. The build system works as intended for now, but can be expanded to use more sophisticated switches to build according to different needs (dev, tests, prod etc).
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: 'works as intended for now' these words seem unnecessary. Saying it can be expanded would itself convey that currently it is restricted to our use case.

File: docs/technology_stack.adoc [hunk 21]
```diff
@@ -7,35 +7,45 @@
...
-The *_src/assets/stylesheets/_* directory includes a *_shared_* directory. These are
-shared global styles which will be refactored into a shared library at some point.
-You can update these styles only if you are making a truly global style, and are going to
-synchronize your changes across all the various UI projects.
+Test Framework::
+Our test stack is built with *Karma*, *Protractor* and *Google Chrome*. *Google Chrome* provides the browser engine to run the app over link:https://www.x.org/archive/X11R7.6/doc/man/man1/Xvfb.1.xhtml[*Xvfb*]. *Protractor* interacts with it as a *WebDriverJS* wrapper and runs the tests according to *Karma's* configurations.
 
-== Gulp
-<TBD>
+Each `*_filename.ts_*` file is supposed to have unit tests along with them as a `*_filename.spec.ts_*` in the same parent directory. The functional tests reside inside `*_./runtime/tests/_*`. For more details about running the tests, please look into link:./testing.adoc[Testing Documentation].
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: Documentation>documentation

File: docs/technology_stack.adoc [hunk 22]
```diff
@@ -7,35 +7,45 @@
...
+
+CI/CD::
+The CI/CD pipeline is made with *Jenkins*, as defined in the `*_./Jenkinsfile_*` at the root of the repo. It uses the content of `*_./deploy_*` (more specifically, `*_./deploy/release.groovy_*`) to orchestrate the CI/CD.
+
+Release Framework::
+We use semantic-release for versioning and publishing our *Planner* library to npm registry. It's fairly configuration less CLI task (`$ npm run semantic-release`) which takes the latest PR, merges and commit messages into account. It generates the new version number, publishes to npm registry, and then generates the release tag on *GitHub* in one simple process.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: It's fairly configuration less>It is a fairly configuration-less 
PR>Pull Request

File: docs/technology_stack.adoc [hunk 23]
```diff
@@ -7,35 +7,45 @@
...
+
+Deployment::
+At the end of the day *Planner* is a static library, and can be used or served with minimal static frontend runtime as a standalone app or with a full fledged Fabric8 UI. So any deployment stack that can make those setups run, can essentially be the deployment platform for *Planner*.
+
+We use *Docker Compose* to generate the image, *Docker* for running the built app container, and *nginx* as the web server to serve the app contents.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: Using Bulleted list would help to spell this out clearly and give it emphasis while skimming for information.
We use:
...

File: docs/technology_stack.adoc [hunk 24]
```diff
@@ -7,35 +7,45 @@
...
+
+== Application Framework
+We use *`Angular 4`* to develop *Planner*. Main application code resides in `*_./src_*`, in which the `*_./src/app_*` contains all the application logic, subdivided into specific functional groups. For the minimalistic runtime added to make *Planner* work standalone, the source lies inside `*_./runtime/src/app_*` which is also an *`Angular 4`* app in itself.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: How about rewording this as:
Location of source code:
- `*_./src_*`  houses the main application code
- `*_./src/app_*` contains all the application logic, subdivided into specific functional groups
- `*_./runtime/src/app_*` contains the source code for the minimalistic runtime which enables the *Planner* standalone. This is also an *`Angular 4`* app in itself.

File: docs/technology_stack.adoc [hunk 25]
```diff
@@ -7,35 +7,45 @@
...
+
+== Programming Languages
+For development ease, we do not use HTML, CSS, JS as is, but Angular Templates, Less and TypeScript instead. For UI elements and components, we use link:http://www.patternfly.org/[PatternFly v3].
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: I would emphasize what we use before what we don't, form the user's pov I want to get straight to the meat-
Suggest: we use Angular Templates, Less and TypeScript  instead of HTML, CSS and JS as is.

File: docs/technology_stack.adoc [hunk 26]
```diff
@@ -7,35 +7,45 @@
...
+
+This requires us to pre-process and build the source-codes into consumable package, but that is automated and this overall accelerates the development process and eases code maintenance.
```
Comment (Preeticp) [inline docs/technology_stack.adoc]: No hyphen in Preprocess, it is one word.
Source code are 2 words, no hyphen.

**Comment by Preeticp** [comment]
@debloper most of my comments are documentation standard or convention specific. I could raise a PR with these changes to your branch if you wish. Let me know how you wish to close this.

**Comment by nimishamukherjee** [comment]
@Preeticp @debloper the changes/queries from my end have been addressed/answered hence approving the PR.

**Comment by debloper** [comment]
@Preeticp regarding the standard and doc conventions, I'd suggest we go ahead with merging the base content in this PR and then you can raise another PR to fix them as it'll be more effective than me fixing them. It'll also clear me off of this PR for the time being. Thanks! :D

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-22 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by Preeticp** [comment]
Makes sense @debloper

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-23 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2069-25 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2069-fabric8-planner.badger.fabric8.io
